File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        cache = {}
        
        for i in range(len(nums)):
            n = nums[i]
            other = target - n
            
            if other in cache:
                return [i, cache[other]]
                
            cache[n] = i
            
        

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hashMap = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int leftover = target - nums[i];
            
            if (hashMap.containsKey(leftover)) {
                return new int[] {i, hashMap.get(leftover)};
            }
            
            hashMap.put(nums[i], i);
        }
        return new int[]{};
    }
}